petastorm/pytorch.py

```diff
@@ -192,6 +202,9 @@ def __iter__(self):
         if self._batch_acc:
             yield self.collate_fn(self._batch_acc)
 
+        self._in_iter = False
```

Comment:
Could you add an inline comment saying that we purposely didn't not wrap it under a final block?